pages/api/popular.ts

```diff
@@ -1,45 +1,55 @@
 import type { NextApiHandler } from "next"
 import { getToken } from "next-auth/jwt"
+import xss from "xss"
 
-import { config } from "config"
-import { HttpClient } from "lib/HttpClient"
 import { HttpError } from "lib/HttpError"
+import { Asset } from "model/Asset"
+import { makeCollibraService } from "services"
+import { getAssetAttributes } from "services/collibra"
 
-type PopularAsset = Collibra.Asset & Pick<Collibra.NavigationStatistic, "numberOfViews">
+type PopularAsset = Asset & Pick<Collibra.NavigationStatistic, "numberOfViews">
 
 const getPopularAssets = async (
   data: PopularAsset[],
-  authorization: string,
+  r: ReturnType<typeof makeCollibraService>,
   limit: number,
   offset = 0,
 ): Promise<PopularAsset[]> => {
-  const mostViewedStats = await HttpClient.get<Collibra.PagedNavigationStatisticResponse>(`${config.COLLIBRA_BASE_URL}/navigation/most_viewed`, {
-    headers: { authorization },
-    query: { offset: offset * limit, limit, isGuestExcluded: true },
-  })
-
-  const assetsResponse = await Promise.all(
-    mostViewedStats.body?.results.map((stat) => HttpClient.get<Collibra.Asset>(`${config.COLLIBRA_BASE_URL}/assets/${stat.assetId}`, {
-      headers: { authorization },
-    })) ?? [],
+  const { body: { results: stats } } = await r((client) => async () => client.get<Collibra.PagedNavigationStatisticResponse>("/navigation/most_viewed", {
+    query: {
+      limit,
+      offset: offset * limit,
+    },
+  }))()
+
+  const collibraAssets = await Promise.all(
+    stats.map((stat) => r((client) => async (id: string) => client.get<Collibra.Asset>(`/assets/${id}`))(stat.assetId)),
   )
 
-  const result = [
+  const assets = collibraAssets
+    .map((res) => res.body)
+    .filter((asset) => asset.type.name.toLowerCase() === "data product")
```

Comment:
Isn't this string "data product" hard coded several times in the code base at this point? Should we add a variable? 🤔

Reply:
two times, but yeah we should probably have a `isDataProductAssetType` utility function

Reply:
Let's leave that as a future enhancement 👍